apps/api/src/app.gateway.ts

```diff
@@ -0,0 +1,44 @@
+import {
+  WebSocketGateway,
+  WebSocketServer,
+  WsException,
+} from '@nestjs/websockets';
+import { Server, Socket } from 'socket.io';
+import { CACHE_MANAGER, Inject, Logger } from '@nestjs/common';
+import { Cache } from 'cache-manager';
+import { AuthService } from './auth/auth.service';
+
+@WebSocketGateway()
+export class AppGateway {
+  @WebSocketServer() server!: Server;
+
+  constructor(
+    private readonly logger: Logger,
+    @Inject(CACHE_MANAGER) private readonly cacheManager: Cache,
+    private readonly authService: AuthService,
+  ) {}
+
+  async handleConnection(client: Socket): Promise<void> {
+    if (!client.handshake.headers?.authorization)
+      throw new WsException('No Authorization header found');
+    const currentUser = this.authService.verifyUserFromToken(
+      client.handshake.headers.authorization,
+    );
+    if (!currentUser) throw new WsException('Invalid token');
+    client.data = { id: currentUser.sub, name: currentUser.name };
+    client.join(`user_${currentUser.sub}`);
+    this.server
+      .to(`user_${currentUser.sub}`)
+      .emit(JSON.stringify({ event_type: 'connected' }));
+    this.cacheManager.set(`user:${currentUser.sub}`, currentUser, 0);
+    this.logger.log(`${currentUser.sub} connected`);
+  }
+
+  async handleDisconnect(client: Socket): Promise<void> {
+    this.cacheManager.del(`user:${client.data.id}`);
```

Comment:
The user maybe has other websockets instances, so they will not be disconnected. This should be checked.